### ISSUE TITLE
Unable to publish events to connections that were dynamically joined to a channel from a hook using context.params.connection.

### ISSUE THREAD

**Issue by petersaints** [feathersjs/feathers#1570]
I know that joining a channel from a hook is not ideal in most situations because it doesn't when running multiple application instances. However, under my usage scenario it makes sense because channels are meant to reflect which connections are currently opened and do not directly reflect persistent information. Moreover, this approach was working before I upgraded to Feahters v4.

I was previously able to make create call to a service which would at some point call a hook that would put the calling connection into a specific channel based on the results of this call by doing:
`context.app.channel(`clients/${context.result.client._id}`).join(context.params.connection);`

And then when I was about to publish an event I could check if the returned data belonged to a certain client and narrow down the channel to `return app.channel(`clients/${data.client._id}`);` However, this stopped working after I moved to Feathers v4. I inspected as much as I could, and while `app.channel(`clients/${data.client._id}`);` appears to select the correct connection, it doesn't work for some reason.

I get the events if I do `return app.channel(app.channels);` but this I really wanted to filter who is able to get the event based on these parameters to avoid notifying everyone about everything, even if they are not interested on those events. Besides, this was also a way to ensure that certain events would not be eavesdropped by people that were not meant to see them.

### System configuration

Tell us about the applicable parts of your setup.

**Module versions** (especially the part that's not working):
```
    "@feathersjs/authentication": "^4.3.2",
    "@feathersjs/authentication-client": "^4.3.2",
    "@feathersjs/authentication-local": "^4.3.2",
    "@feathersjs/authentication-oauth": "^4.3.2",
    "@feathersjs/configuration": "^4.3.2",
    "@feathersjs/errors": "^4.3.2",
    "@feathersjs/express": "^4.3.2",
    "@feathersjs/feathers": "^4.3.2",
    "@feathersjs/primus": "^4.3.2",
    "@feathersjs/socketio": "^4.3.2",
    "@feathersjs/socketio-client": "^4.3.2",
    "feathers-authentication-hooks": "^0.3.1",
    "feathers-hooks-common": "^4.20.7",
    "feathers-mongodb": "^5.0.0",
    "feathers-mongoose": "^8.0.2",

```

**NodeJS version**: 10.16.3

**Operating System**: Ubuntu 19.04 (amd64)

**Browser Version**: Chrome Version 78.0.3904.21 (Official Build) beta (64-bit)

**React Native Version**: Not applicable

**Module Loader**: Webpack on the client side

**Comment by daffl** [comment]
I don't think anything in the channel system changed so this should still be the same. Do you have a reproducible example that shows your issue?

**Comment by petersaints** [comment]
@daffl It's hard to come up with a self-contained example for this issue. The best I think I could do is try to create an empty Feathers project and try to reproduce the issue there. Would that work for you?

**Comment by daffl** [comment]
Yes, that's what I would suggest.

**Comment by petersaints** [comment]
Ok. I'll try to do that when I have some free time and I'll get back to
you.

**Comment by daffl** [comment]
Closing since the issue has been open for a while without any new information. Please feel free to reopen with more information at any time 😄

**Comment by petersaints** [comment]
@daffl 
I eventually found an alternative way of managing my channels that for the time being is acceptable for my project. However, I may eventually need to join channels from hooks once more. 

So I decided to make an empty project without authentication and with custom blank services created by the Feathers CLI. So far, I have not been able to reproduce the issue.

However, if I run into this problem again on any project I will try to to mimic what's happening on a toy example, share it with you and ask you to reopen the issue (I don't think I can do it myself).
